fflas-ffpack/ffpack/ffpack_sss.inl

```diff
@@ -0,0 +1,513 @@
+        typename Field::Element_ptr Temp2 = FFLAS::fflas_new(Fi, s, s);
+
+            /* Row 2 */
+        if (kf > 1)
+                /* A_{2, 1} <- P_{2} * Q_{1} */
+            fgemm (Fi, FFLAS::FflasNoTrans, FFLAS::FflasNoTrans, s, s, s, Fi.one, P, 
+                   ldp, Q, ldq, Fi.zero, A +  s * lda, lda);
+    
+            /* After row 2, R is also applied */
+        for (size_t row = 1; row < kf - 1; row++) // Loop on rows which have LT part
+        {
+                /* Loop needs Temp1, which is first set here */
+                /* A_{row + 2, row + 1} <- P_{row + 2} * Q_{row + 1} */
+            fgemm (Fi, FFLAS::FflasNoTrans, FFLAS::FflasNoTrans,
+                   s, s, s, Fi.one, P + row * s * ldp, // In this loop, all blocks are s x s
+                   ldp, Q + row * s * ldq, ldq, Fi.zero, A + (row + 1) * s * lda + row * s, lda);
+                /* Temp1 <- P_{row + 2} * R_{row + 1} */
+            fgemm (Fi, FFLAS::FflasNoTrans, FFLAS::FflasNoTrans, s, s, s, Fi.one, P + row * s * ldp, 
+                   ldp, R + (row - 1) * s * ldr, ldr, Fi.zero, Temp1, s);
+        
+                /* Instructions are doubled in the loop in order to avoid using more than two temporary blocks */
+            for (size_t block = 1; block < row; block+=2)
+            {
+                    /* A_{row + 2, row + 1 - block} <- Temp1 * Q_{row + 1 - block} */
+                fgemm (Fi, FFLAS::FflasNoTrans, FFLAS::FflasNoTrans, s, s, s, Fi.one, Temp1, s, 
+                       Q + (row - block) * s * ldq, ldq, Fi.zero, A + (row + 1) * s * lda + (row - block) * s, lda);
+                    /* Temp2 <- Temp1 * R_{row + 1 - block} */
+                fgemm (Fi, FFLAS::FflasNoTrans, FFLAS::FflasNoTrans, s, s, s, Fi.one, Temp1, 
+                       s, R + (row - 1 - block) * s * ldr, ldr, Fi.zero, Temp2, s);
+
+                    /* A_{row + 2, row - block} <- Temp2 * Q_{row - block} */
+                fgemm (Fi, FFLAS::FflasNoTrans, FFLAS::FflasNoTrans, s, s, s, Fi.one, Temp2, s,
+                       Q + (row - block - 1) * s * ldq, ldq, Fi.zero, A + (row + 1) * s * lda + (row - block - 1) * s, lda);
+                    /* If necessary:
+                     * - Not last loop
+                     * - One more block
+                     * -> At least one more block */
+                if (block + 1 < row)
+                        /* Temp1 <- Temp2 * R_{row - block} */
+                    fgemm (Fi, FFLAS::FflasNoTrans, FFLAS::FflasNoTrans, s, s, s, Fi.one, Temp2, 
+                           s, R + (row - block - 2) * s * ldr, ldr, Fi.zero, Temp1, s);
+            }
+                /* First column if not done already */
+            if (row%2)
+                    /* A_{row + 2, 1} <- Temp1 * Q_{1} */
+                fgemm (Fi, FFLAS::FflasNoTrans, FFLAS::FflasNoTrans, s, s, s, Fi.one, Temp1, s,
+                       Q, ldq, Fi.zero, A + (row + 1) * s * lda, lda);
+        }
+    
+            /* Last row if partial */
```

Comment:
Maybe merge this part within the main loop, setting the dimension to rs for the last iteration.

Reply:
Fixed